plugins/outputs/mqtt/topic_parser.go

```diff
@@ -0,0 +1,34 @@
+package mqtt
```

Comment:
This isn't parsing actually but rather _constructing_ the topic. You might also want to keep this single function in `mqtt.go` but this is not a must.

Reply:
Sure, I can do that. I don't have much experience with Go and here I am open to recommendations.
I feel that SRP moves me to think we need a different abstraction to handle the parsing or construction of the topic string.
If I embedded this function inside the main one, I am not sure how to keep the tests simple if I want to test via the public interface of the main package.

If you move it inside the main package, what testing strategy do you recommend? I am not sure if we use mocks or spies in Go, where we can check if a certain method has been called with certain parameters.